lib/srv/db/redis/engine.go

```diff
@@ -584,15 +586,20 @@ func isTeleportErr(err error) bool {
 // driverLogger implements go-redis driver's internal logger using logrus and
 // logs everything at TRACE level.
 type driverLogger struct {
-	*logrus.Entry
+	*slog.Logger
 }
 
-func (l *driverLogger) Printf(_ context.Context, format string, v ...any) {
-	l.Entry.Tracef(format, v...)
+func (l *driverLogger) Printf(ctx context.Context, format string, v ...any) {
+	if !l.Logger.Enabled(ctx, logutils.TraceLevel) {
```

Comment:
Is this to avoid unecessary `fmt.Sprintf`?

Reply:
no i don't have any good idea either.

we can put format/v in a struct (something like FormattedMessage) and implement `slog.LogValuer` similar to other wrappers we have in log util but not sure if it's worth it